api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -310,3 +310,26 @@ message Principal {
     Principal not_id = 8;
   }
 }
+
+// Action defines the result of allowance or denial when a request matches the matcher.
+message Action {
+  // The name indicates the matcher name.
+  string name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The action to take if the matcher matches. Every action either allows or denies a request,
+  // and can also carry out action-specific operations.
+  //
+  // Actions:
+  //
+  //  * ALLOW: Allows the request if and only if the matcher that matches the
+  //    request.
+  //  * DENY: Allows the request if and only if the matcher that does not match
+  //    the request.
+  //  * LOG: Allows the request if and only if the matcher that matches the
```

Comment:
Rephrase please. Is the hint set to false by default without any LOG actions?

Reply:
No, when the whole matcher tree contains at least one LOG action, the hint can be set to true or false. I may change to
```proto
//  * LOG: Allows the request if and only if the matcher that matches the
//    request. When the whole matcher trees contains at least one LOG action,
//    the dynamic metadata key `access_log_hint` under the shared key
//    namespace 'envoy.common' will be set to the value `true` if the matcher
//    matches the request, and set to the value `false` if the matcher does
//    not match.
```
Do you think it is good?

Reply:
I think it would be more clear if we split this into two parts. First, explain that the matcher may or may not produce an action. If it produces an action, it is unique (is it?). If it doesn't produce an action, what happens? Then, for each action, it's pretty self-explanatory I think what Allow/Deny/Log do.

Reply:
One aspect that might be confusing: can there be more than one action at a time? It would make sense to have Log and Deny at the same time.

Reply:
> One aspect that might be confusing: can there be more than one action at a time? It would make sense to have Log and Deny at the same time.

No. In fact, the policy approach also does not provide log and deny.

Reply:
> If it produces an action, it is unique (is it?).

I am not sure what you means unique. Every matcher (input, matcher, action pair) requires an action. Any request get matched will have an action. Any request that does not get matched will fallback to DENY.

Reply:
Ack. My main concern is just to make it easy to read. "Allows the request if and only if the matcher that does not match the request" -> If the matched action is Allow, request is permitted, etc.

Reply:
Ack, for someone new to this API, it's probably important to know that matching is done sequentially looking for the first action. Policy languages are often declarative. I think having a good example in the docs translating the old style to this style would probably do a good job communicating the new API.

Reply:
```proto
// Actions:
//
//  * ALLOW: If the request gets matched on ALLOW, it is permitted.
//  * DENY: If the request gets matched on DENY, it is not permitted.
//  * LOG: If the request gets matched on LOG, it is permitted. Besides, the
//    dynamic metadata key `access_log_hint` under the shared key namespace
//    'envoy.common' will be set to the value `true`.
//  * If the request cannot get matched, it will fallback to DENY.
//
// Log behavior:
//
//  If the RBAC matcher contains at least one LOG action, the dynamic
//  metadata key `access_log_hint` will be set based on if the request
//  get matched on the LOG action.
```

Maybe separate the action and log behavior, and emphasize the case not getting matched. Do you think it looks better?

Reply:
Yeah, this is fine now.